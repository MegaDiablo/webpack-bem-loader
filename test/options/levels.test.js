const webpack = require('../helpers/compiler');

describe('Options', () => {
    describe('levels', () => {
        test('common/desktop', async () => {
            const mock = {
                'index.js' : `require('b:button')`,
                'common.blocks' : {
                    'button' : {
                        'button.js' : `({ block: 'button', content: 'common' })`
                    }
                },
                'desktop.blocks' : {
                    'button' : {
                        'button.js' : `({ block: 'button', content: 'desktop' })`
                    }
                }
            };
            const config = {
                loader : {
                    test : /\.js$/,
                    options : {
                        levels : [
                            'common.blocks',
                            'desktop.blocks'
                        ]
                    }
                }
            };

            const { stats } = await webpack('index.js', { config, mock });
            const { source } = stats.toJson().modules[1];

            expect(source).toMatchSnapshot();
        });
    });
});
