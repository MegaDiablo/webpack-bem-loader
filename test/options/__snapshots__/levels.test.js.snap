// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options levels common/desktop 1`] = `
"[(
require('./common.blocks/button/button.js'),
(require('./desktop.blocks/button/button.js').default || require('./desktop.blocks/button/button.js')).applyDecls()
)][0]"
`;
